### DISCUSSION TITLE
What are placeholder units? Under what circumstances can they be requested?

### DISCUSSION THREAD

**Discussion by SunlitZelkova** [PygmalionOfCyprus/cmo-db-requests#1264] [Q&A] [answered]
Hello,

In the DB3K, the Chinese Type 003 aircraft carrier is available for use in the game, but has "placeholder" written as a comment attached to its name in the DB. Close inspection makes it appear as though it has very similar weapons to that of the Type 002 Shandong carrier.

What is a "placeholder"? I would assume this is when a unit is added despite a lack of clear information of its characteristics, and just has stats borrowed from a past unit. I am asking to be sure.

A similar situation seems to have occurred with the Type 055 destroyer as long ago in the days of CMANO, the Type 055 had near identical weapons and sensors to the Type 052D which was actually in service at the time, when the first Type 055 had yet to launch. Although I have not checked, it is probably safe to assume that the Type 095 SSGN present in the DB is also a placeholder, as the Type 095 has not been revealed yet and no one even knows what it looks like.

And then a second question- under what circumstances can a unit be requested for in placeholder form?

What I have in mind to make a request for is the H-20 bomber and Type 096 SSBN, two vehicles no one knows even knows the appearance of. We have estimates from Western intelligence agencies and a general understanding of what weapons could be expected (for example, the CJ-10 ALCM for the H-20 and the new JL-3 SLBM for the Type 096), but we can't make good guesses about performance.

I would like to request these units despite the lack of info on them, as I am making a campaign set in the late 2020s when these would be in service.

Another unit this "placeholder" method could be applied to (and which I would also request if it qualifies) is the B-21 Raider.

**Comment by PygmalionOfCyprus** [comment, marked as answer]
It's a good question.

Generally, I'm loathe to add placeholder units. I don't think they have very much utility, especially for the potential DB bloat they introduce. (You can imagine, for example, if I were to create placeholders for every proposed Chinese/Russian/DARPA/etc. program how the DB would quickly become a graveyard of cancelled concepts.) If I'm going to just make a carbon copy of something already in the DB and put "PLACEHOLDER" in the comments, people may as well just use those base units directly and rename them to fit their scenario.

With that said, I think there's some merit to adding placeholders for high-profile unknowns that people are likely to want for scenarios and that we can confidently say _are_ one day going to emerge, the logic being that a future DB update can replace those placeholders with real(er) capabilities without requiring that units be deleted and re-added, triggers adjusted, LUA changed, etc. The Type 003 is immediately relevant to near-future operations in the Pacific and not likely to be scrapped at this stage. The B-21 has obvious implications for American power projection and weapons delivery and likewise is due to appear in public shortly. Et cetera, et cetera.

I would be a little bit more cautious about things like the H-20. The extent of my knowledge on that particular program goes no further than the Wikipedia article, so I don't know how reliably we can count on it entering service, but if I had a nickel for every STEALTH BOMBER THAT CAN THREATEN AMERICA supposedly DUE TO ENTER SERVICE THIS YEAR I'd be able to buy one of my own. If we have reason to believe it's truly likely to emerge for real then we can consider giving it a placeholder.

Of course, in cases where we _aren't_ sure of imminent deployment but still want to add it because it creates an interesting capability/problem we can always go the good ol' route of adding something as a hypothetical.
